### PR TITLE
Fix equals and implement missing ItemStackMock methods

### PR DESCRIPTION
**Pull request by alexcrea**
# Description
Another fix for ItemStackMock
- Fix equals function
- Implement most part or throw exception on method referencing delegated ItemStack. (null on ItemStackMock)
- Also ensure AIR item has not ItemMeta after setItemMeta

Also implemented PersistentDataContainerViewMock as it is required for getPersistentDataContainer in ItemStack

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added (if applicable).

### CODE REVIEW

File: src/test/java/org/mockbukkit/mockbukkit/persistence/PersistentDataContainerViewMockTest.java
Comment (Thorinwasher) [inline src/test/java/org/mockbukkit/mockbukkit/persistence/PersistentDataContainerViewMockTest.java:1]: This file does not follow our style guidlines

**Review by Thorinwasher — CHANGES_REQUESTED** [review]
It looks good to me except for code style in one file. Sorry for the large delay in reviewing this by the way.

File: src/main/java/org/mockbukkit/mockbukkit/inventory/ItemStackMock.java [hunk 1]
```diff
@@ -282,7 +356,7 @@ public boolean equals(Object obj)
 		}
 		if (stack instanceof ItemStackMock bukkit)
 		{
-			return isSimilar(bukkit) && this.amount == bukkit.getAmount() && this.durability == bukkit.durability && Objects.equals(this.getLore(), bukkit.getLore()) && Objects.equals(this.getEnchantments(), bukkit.getEnchantments());
+			return isSimilar(bukkit) && this.amount == bukkit.getAmount() && this.durability == bukkit.durability && Objects.equals(this.itemMeta, bukkit.getItemMeta());
```
Comment (Insprill) [inline src/main/java/org/mockbukkit/mockbukkit/inventory/ItemStackMock.java:359]: Does this match Paper's impl? I remember it being weird and not actually checking if *everything* is equal, not sure if they changed that at some point.

Comment (Thorinwasher) [reply src/main/java/org/mockbukkit/mockbukkit/inventory/ItemStackMock.java:359]: I checked this, and it seems to check everything now

Comment (alexcrea) [reply src/main/java/org/mockbukkit/mockbukkit/inventory/ItemStackMock.java:359]: Thanks for checking !

File: src/test/java/org/mockbukkit/mockbukkit/persistence/PersistentDataContainerViewMockTest.java
Comment (alexcrea) [reply src/test/java/org/mockbukkit/mockbukkit/persistence/PersistentDataContainerViewMockTest.java:1]: oups

**Review by Thorinwasher — APPROVED** [review]

**Pull request merged**